packages/util/src/address.ts

```diff
@@ -37,6 +38,14 @@ export class Address {
     return new Address(zeros(20))
   }
 
+  /**
+   * Returns an Address object from a bigint address (they are stored as bigints on the stack)
+   * @param value The bigint address
+   */
+  static fromBigInt(value: bigint): Address {
+    return new Address(setLengthLeft(bigIntToBytes(value & MASK_160), 20))
```

Comment:
I don't think we should put the mask here. This has the undesired side effect that if an user passes a bigint > 20 bytes, it does not throw but silently throws away the upper bytes. (We should thus do this explicitly in EVM, but not in this method). This method should thus also throw if you pass a bigint > 20 bytes.

Reply:
However, it should still `setLengthLeft` after checking that the bigint <= 20 bytes. So you can `Address.fromBigInt(BigInt(1))` without problems.

Reply:
Ok thanks for the note. So you are suggesting that we:
- Add length validation to the `fromBigInt` helper so that it throws if length > 20
- Remove the bitwise mask operation and instead do it in usage. 
- Keep setLengthLeft(value,20) as we do now

Since fromBigInt is strictly used (in our monorepo) for the evm, I'm not sure how to best architect this. Will experiment

Reply:
Yes, exactly as you have noted here. So the EVM should call `fromBigInt` with the masked variant, `fromBigInt` should throw if bytes > 20 and it should always perform the `setLengthLeft` (otherwise Address constructor will throw on bytes < 20 bytes)

Reply:
I made the adjustment above and decided to create a helper method in the evm that does the masking, curious to hear what you think!